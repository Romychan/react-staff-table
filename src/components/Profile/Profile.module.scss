.container {
  @apply flex items-center text-gray-900 whitespace-nowrap dark:text-white;
}

.avatar {
}

.content {
  @apply ml-4;
}

.name {
  @apply text-base font-semibold;
}

.username {
  @apply font-normal text-gray-500 dark:text-gray-400;
}
