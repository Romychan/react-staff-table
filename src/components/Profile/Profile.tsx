import { Avatar } from '@ui/Avatar/Avatar';

import styles from './Profile.module.scss';

interface IProfileProps {
  name: string;
  username: string;
  imageSrc: string;
}

export const Profile = ({ name, username, imageSrc }: IProfileProps) => {
  return (
    <div className={styles.container}>
      <Avatar src={imageSrc} alt={name} />
      <div className={styles.content}>
        <div className={styles.name}>{name}</div>
        <div className={styles.username}>{username}</div>
      </div>
    </div>
  );
};
