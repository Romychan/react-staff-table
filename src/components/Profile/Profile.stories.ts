import type { Meta, StoryObj } from '@storybook/react';

import { Profile } from './Profile';

/** This component is used to display the users avatar with his name and username. */
const meta = {
  title: 'Components/Profile',
  component: Profile,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof Profile>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    name: 'First Name',
    username: 'username1',
    imageSrc:
      'https://robohash.org/doloresmollitiaaperiam.png?size=50x50&set=set1',
  },
};
