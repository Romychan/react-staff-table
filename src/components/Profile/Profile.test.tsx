import { RenderResult, render, screen } from '@testing-library/react';

import { Profile } from './Profile';

describe('Profile', () => {
  let container: RenderResult;

  beforeEach(() => {
    container = render(
      <Profile
        name="Test Name"
        username="Test Username"
        imageSrc="./test.png"
      />,
    );
  });

  it('should render correctly', () => {
    const profileElement = screen.getByText(/Test Username/i);
    const imageElement = screen.getByAltText(/Test Name/i);

    expect(profileElement).toBeInTheDocument();
    expect(imageElement).toHaveAttribute('src', './test.png');
  });

  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
