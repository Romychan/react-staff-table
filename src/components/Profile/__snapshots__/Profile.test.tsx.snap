// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Profile > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_container_a5316a"
      >
        <img
          alt="Test Name"
          class="_avatar_3ecbb0"
          src="./test.png"
        />
        <div
          class="_content_a5316a"
        >
          <div
            class="_name_a5316a"
          >
            Test Name
          </div>
          <div
            class="_username_a5316a"
          >
            Test Username
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_container_a5316a"
    >
      <img
        alt="Test Name"
        class="_avatar_3ecbb0"
        src="./test.png"
      />
      <div
        class="_content_a5316a"
      >
        <div
          class="_name_a5316a"
        >
          Test Name
        </div>
        <div
          class="_username_a5316a"
        >
          Test Username
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
